sway/tree/workspace.c

```diff
@@ -730,19 +691,39 @@ void workspace_add_gaps(struct sway_workspace *ws) {
 			focus = seat_get_focus_inactive_view(seat, &focus->node);
 		}
 		if (focus && focus->view && view_is_only_visible(focus->view)) {
+			ws->current_gaps.top = 0;
+			ws->current_gaps.right = 0;
+			ws->current_gaps.bottom = 0;
+			ws->current_gaps.left = 0;
 			return;
 		}
 	}
 
 	ws->current_gaps = ws->gaps_outer;
-	if (ws->layout == L_TABBED || ws->layout == L_STACKED) {
-		// We have to add inner gaps for this, because children of tabbed and
-		// stacked containers don't apply their own gaps - they assume the
-		// tabbed/stacked container is using gaps.
-		ws->current_gaps.top += ws->gaps_inner;
-		ws->current_gaps.right += ws->gaps_inner;
-		ws->current_gaps.bottom += ws->gaps_inner;
-		ws->current_gaps.left += ws->gaps_inner;
+	// Add inner gaps and make sure we don't turn out negative
+	ws->current_gaps.top = MAX(0, ws->current_gaps.top + ws->gaps_inner);
+	ws->current_gaps.right = MAX(0, ws->current_gaps.right + ws->gaps_inner);
+	ws->
+current_gaps.bottom = MAX(0, ws->current_gaps.bottom + ws->gaps_inner);
```

Comment:
Can you fix this? The line splitting is weird, but it should actually fit on a single line anyway

Reply:
Yep, this was just an errant Enter that went into the text editor instead of the terminal somehow.